src/python/pants/rules/core/test.py

```diff
@@ -121,9 +121,7 @@ class AddressAndDebugRequest:
 
 @union
 class CoverageDataBatch(ABC):
-    @abstractmethod
-    def __init__(self, addresses_and_test_results: Tuple[AddressAndTestResult, ...]) -> None:
-        """Subclasses should accept this in their constructor."""
```

Comment:
Note to other reviewers: An abstract `__init__.py` causes issues for subclasses.

Reply:
Then ABC should be dropped since it does nothing.